### DISCUSSION TITLE
Is it possible to use firebase messaging without analytics? App has been rejected by Apple.

### DISCUSSION THREAD

**Discussion by victoriaSh** [invertase/react-native-firebase#4947] [Q&A] [answered]
Hello,
My app for kids has been rejected 6 times for Safety reasons. 
I use @react-native-firebase/messaging for push-notifications. This package requires @react-native-firebase/app.
By default @react-native-firebase/app has AdSupport, GoogleAppMeasurement and FirebaseAnalytics frameworks in dependencies.

I've already removed AdSupport from pods. But app was rejected : 
"We continue to find that your app implements the App Tracking Transparency framework, which is used when apps collect data about users and share it with third-parties for tracking purposes." 

I haven't found any ATT framework but I found that GoogleAppMeasurement.framework matches 'NSUserTrackingUsageDescription' which can be related to App Tracking Transparency.

Is it possible to use only firebase-messaging without GoogleAppMeasurement, FirebaseAnalytics or any other code violating kids category guideline?

**Comment by mikehardy** [comment, marked as answer]
Hi there - without information on versions I can't say. I suspect you are using old versions somehow, as `@react-native-firebase/app` only uses Firebase-CoreOnly which is supposed to have no IDFA-containing transitive dependencies

https://github.com/invertase/react-native-firebase/blob/e173954908e6dc4a7bad6b7d1c6a8a06c2d76756/packages/app/RNFBApp.podspec#L30

Have you searched issues for past information? Others have been in this area if I recall

I have tested combinations of the react-native-firebase without IDFA in order to probe iOS builds and get a result that was okay for kids apps if I recally correctly, you can see the testing rig here and run it yourself to see how it builds / what it includes:

https://github.com/mikehardy/rnfbdemo/blob/37b13cd0ea36a2db650c63d365d3bc11b7a084c7/make-demo.sh#L9-L14

At the time I got IDFA-free builds with everything but admob? https://github.com/mikehardy/rnfbdemo/blob/37b13cd0ea36a2db650c63d365d3bc11b7a084c7/make-demo.sh#L142-L144

In general since this is a very subtle area and has evolved quickly you should very carefully check the firebase-ios-sdk upstream docs as well:

**Reply by victoriaSh** [reply]
Thank you so much for your response! It helped me a lot. I've updated @react-native-firebase/app(8.4.6) and @react-native-firebase/messaging(7.9.1) to the latest versions. Also found "pod 'Firebase/Analytics'" in Podfile and removed it. And app was already approved just in 15 minutes without any additional questions.

It seems that rules for reviewing change very quickly because there was no such problem a couple of weeks ago.

**Reply by mikehardy** [reply]
I have had a similar experience - but not so much that the rules change but that they are applied haphazardly - you just never know whether the Apple Reviewer is going to look at everything (or anything!) or not, and you can just be rejected for legitimate reasons even though they accepted them days earlier even. It's kind of annoying actually, but... what can you do :-)

Really glad to hear you've got your update approved, good luck with your project
